### DISCUSSION TITLE
stbi_load doesn't work with jpg, data is null

### DISCUSSION THREAD

**Discussion by Mordentary** [nothings/stb#1364] [Q&A] [answered]
First of all, here's what I already tried: different paths - relative/absolute, checked the JPEG encoding format (it was always baseline), opened pictures through fopen (they open), and of course tried different pictures. Also tried different versions of stb. Always data is null and always stbi_failure_reason gives this error: "unknown image type". At the same time, it works perfectly with png format. There are several images here that I used:
https://imgur.com/GZAF6A0
https://imgur.com/MlRyuJk
https://imgur.com/73YGD6N
Code:
https://pastebin.com/5aV5WEGG

**Comment by nothings** [comment, marked as answer]
Tried the first one (in my own codebase, not your test code), it loaded fine for me.

What platform are you on and/or what compiler are you using?

You could try using fopen to load the file yourself (make sure you open in binary mode), and then use `stbi_load_from_memory` and see if that's any different, but I can't imagine why it would be.

If you're willing to debug it, breakpoint in `stbi__err()`. Then run until the callstack shows the jpeg decoder (possibly other image formats will be attempted first and generate spurious calls to `stbi__err`). Then look up one level on the callstack and see what's calling one of the `stbi__err*()` macros.

**Reply by nothings** [reply]
I have updated the previous answer several times, and am posting this message since you won't get notified of those updates.

**Reply by Mordentary** [reply]
Windows 10/MSVC. I'll try everything you listed and come back here with the results, thanks.

**Reply by Mordentary** [reply]
Okay, I figured out what the problem is. If I directly define implementation in the class where I load the texture, then everything works fine. I have one project referencing to another.. Here's what the solution structure looks like:
https://imgur.com/5kqTAL8

I'm wondering why png files loaded fine...

**Reply by Mordentary** [reply]
I understand that all this strange behavior is due to a conflict between assimp and your lib. Thanks, good luck!

**Reply by nothings** [reply]
It appears assimp uses stb internally, so probably the cleanest way to address this is to define STB_IMAGE_STATIC when including it with STB_IMAGE_IMPLEMENTATION, which makes all the definitions private to the file. but really assimp should do this to avoid causing collisions with clients.
